[lex.separate] Redistribute second comment (#7351)
cplusplus/draft: source/lex.tex

called a \defnadj{preprocessing}{translation unit}.
\begin{note}
A \Cpp{} program need not all be translated at the same time.

> Translation units can be separately translated and then later linked to
> produce an executable program\iref{basic.link}.

Reviewer: ```suggestion
Translation units can be separately translated and then linked together to
produce a program\iref{basic.link}.
```
This change would bring the note more in line with the choice of words in https://eel.is/c++draft/basic.link#1

Author: For now I was deliberately restricting myself to moving approved words around.

I have no objections to further refining the existing text, but would rather do that as a follow-up once the text is in its new place.

Author: For the record, my own rewording would be more like:
"Source files can be separately translated to produce translation units and instantiation units that are later linked to produce an executable program."

Reviewer 2: side note: Of course, "instantiation units" no longer exist, really, given that we (must) have interleaved instantiation and constant evaluation.

Reviewer 3: Are we still OK with this move then?

Reviewer 3: Or should we wait for Reflection to land first?

Author: I think this would have made a good addition for N5001, so that the reflection folks could have had this slightly cleaner base to work from.  As it missed P5001, I suggest we refer the authors of P2996 to look at this and pick up anything they find useful.
